### DISCUSSION TITLE
Can not supprot stream queue with SAC

### DISCUSSION THREAD

**Discussion by LucasZhanye** [rabbitmq/amqp091-go#134] [General]
I use Rabbitmq3.11.3, and the official say "Single active consumer for streams is a feature available in RabbitMQ 3.11 and more."
but when I use amqp091-go  to declare a queue which type is steam,just like this:
queue, err := channel.QueueDeclare(
		"q5",
		true,
		false,
		false,
		false,
		amqp091.Table{
			"x-single-active-consumer": true,
			"x-queue-type":             "stream",
		},
	)
it always report error： 2022/11/23 16:56:19 Failed to declare a queue: Exception (406) Reason: "PRECONDITION_FAILED - invalid arg 'x-single-active-consumer' for queue 'q5' in vhost '/' of queue type rabbit_stream_queue"
How can I do for it?

**Comment by Zerpet** [comment]
AMQP single active consumer and Stream single active consumer are two different features. To use Stream SAC, you have to use one of the Stream clients that implement the native stream protocol. The bridge of AMQP to Stream does not support this feature. You can read more about Stream SAC here:

https://blog.rabbitmq.com/posts/2022/07/rabbitmq-3-11-feature-preview-single-active-consumer-for-streams/

At the moment, the native stream clients that implement stream SAC and Super Streams are the [Java Stream Client](https://github.com/rabbitmq/rabbitmq-stream-java-client) and [Dotnet Stream Client](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client).

**Comment by Gsantomaggio** [comment]
I'd add that we have the [Golang Stream](https://github.com/rabbitmq/rabbitmq-stream-go-client) client, but the SAC feature is not ready yet.  We are open to contributions btw :)

**Comment by LucasZhanye** [comment]
> I'd add that we have the [Golang Stream](https://github.com/rabbitmq/rabbitmq-stream-go-client) client, but the SAC feature is not ready yet. We are open to contributions btw :)
Thanks for your answer!
But I always feel uncertain for it, that  I just only use golang stream client or amqp901 client? Or a mix of the two?

**Comment by Gsantomaggio** [comment]
Please read the documentation about that https://www.rabbitmq.com/stream-core-plugin-comparison.html

**Comment by LucasZhanye** [comment]
> AMQP single active consumer and Stream single active consumer are two different features. To use Stream SAC, you have to use one of the Stream clients that implement the native stream protocol. The bridge of AMQP to Stream does not support this feature. You can read more about Stream SAC here:
> 
> https://blog.rabbitmq.com/posts/2022/07/rabbitmq-3-11-feature-preview-single-active-consumer-for-streams/
> 
> At the moment, the native stream clients that implement stream SAC and Super Streams are the [Java Stream Client](https://github.com/rabbitmq/rabbitmq-stream-java-client) and [Dotnet Stream Client](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client).

Thanks !  Maybe it not suit for my scenario, I want to create some stream queue and some classic queue

**Reply by Zerpet** [reply]
You can use both clients, if you need features specific to AMQP and Stream. Stream Core (Stream over AMQP) is a bridge to facilitate adoption of streams using a known API (AMQP API), at the expense of losing some performance. If you have a requirement to use SAC in classic queues and in Streams, you will have to use AMQP client for classic queues + sac, and Stream client for stream + sac.

**Comment by Zerpet** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(
